src/app/shared/guards/pending-changes-guard.ts

```diff
@@ -44,16 +46,26 @@ export class PendingChangesGuard implements CanDeactivate<PendingChangesComponen
       icon: 'pi pi-exclamation-triangle',
       acceptLabel: $localize`Yes, leave page`,
       accept: () => {
-        dialogResponse.next(true);
-        dialogResponse.complete();
+        dialogResponse.next('accepted');
       },
       rejectLabel: $localize`No`,
       reject: () => {
-        dialogResponse.next(false);
-        dialogResponse.complete();
+        dialogResponse.next('declined');
       },
     });
-    return dialogResponse;
+
+    dialogResponse$.pipe(
+      filter(status => status !== 'pending'),
+      take(1),
+      delay(0),
+    ).subscribe(() =>
+      dialogResponse.next('pending')
```

Comment:
A subject subscription which publishes on itself... that's very smelly. That shows a "invalid" usage of RxJs.